js/services/IRAnalysis.js

```diff
@@ -150,6 +150,27 @@ define(function (require, exports) {
 			});
 	}
 
+	function getSql(expression, analysisId, options) {
+		const data = {
+			expression: expression,
+			analysisId: analysisId,
+			options: options,
+		  };
+		const getSqlPromise = httpService.doPost(config.webAPIRoot + 'ir/sql', data);
+		getSqlPromise.catch((error) => {
+			console.log("Error: " + error);
+			authApi.handleAccessDenied(error);
```

Comment:
What is the logic here? Why access denied handled here but not handled in the `translateSql` below?

Reply:
access denied handler was added to translateSql